core/base/executor.hpp

```diff
@@ -689,6 +689,8 @@ class GpuExecutor : public detail::ExecutorBase<GpuExecutor>,
 
     void synchronize() const override;
 
+    void run(const Operation &op) const override;
```

Comment:
Is this necessary? I thought that `ExecutorBase<GpuExecutor>::run()` implementation is doing the correct thing here...

Reply:
Ah, figured it out - disregard the previous comment.

Reply:
See your first point in #11. AFAIK this is required because run needs to be changed to have two behavior, when actual GPU code is launched we need the device guard to target the proper GPU, and otherwise we launch the kernel which should throw unimplemented.